google/api_core/future/polling.py

```diff
@@ -78,16 +78,16 @@ def done(self, retry=DEFAULT_RETRY):
         # pylint: disable=redundant-returns-doc, missing-raises-doc
         raise NotImplementedError()
 
-    def _done_or_raise(self):
+    def _done_or_raise(self, retry=DEFAULT_RETRY):
         """Check if the future is done and raise if it's not."""
-        if not self.done():
+        if not self.done(retry):
```

Comment:
This should be passed as a keyword argument

```suggestion
        if not self.done(retry=retry):
```

Reply:
On further thought, since this is a new kwarg, we don't want to force all clients that subclass it to add it in order to use the next `google-api-core` (this is a breaking change).

We should really only populate `retry` at all when explicitly passed a value for one.

Maybe something like this?

```suggestion
        done_kwargs = {}
        if retry is not DEFAULT_RETRY:
            done_kwargs["retry"] = retry
        if not self.done(**done_kwargs):
```